### PR TITLE
feat(profile): Adds certificates to static profile path

### LINKED ISSUE #16246: [beta] Certificates should be added to the static profile path
Need to add the new certificate buttons to the static profile before we have the react profiles. This is blocker unless that feature is available.

The code that needs to be updated is here:

https://github.com/freeCodeCamp/freeCodeCamp/blob/5a18d1cf8091da919cd213017b55888ad83d320e/server/views/account/show.jade#L51-L58

The new certificate object ids are here:

https://github.com/freeCodeCamp/freeCodeCamp/blob/5a18d1cf8091da919cd213017b55888ad83d320e/common/models/user.json#L162-L200

### PR DESCRIPTION
**Pull request by cpheimbach**
I added all boolean checks, buttons and routes of the new certificates
to the view. (According to user.json)

Closes #16246

#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #16246

### CODE REVIEW

File: server/views/account/show.jade [hunk 1]
```diff
@@ -56,6 +56,24 @@ block content
                     if isBackEndCert
                         .button-spacer
                         a.btn.btn-primary.btn-block(href='/' + username + '/back-end-certification') View My Back End Development Certification
+                    if isFullStackCert
```
Comment (raisedadead) [inline server/views/account/show.jade]: I forgot to mention that we do not need this particular cert view button here, can you please remove this?

i.e delete lines 59-61?

**Review by raisedadead — CHANGES_REQUESTED** [review]
Can you please address this change?

```
git checkout fix/adds_certificates_to_static_profile_path
```

Make changes

```
git commit --amend --no-edit
git push --force origin fix/adds_certificates_to_static_profile_path
```

File: server/views/account/show.jade (hunk 1 above)
Comment (cpheimbach) [reply server/views/account/show.jade]: sure, will do.

**Comment by raisedadead** [comment]
@cpheimbach looks like you forgot to do a ```git add <file name>``` after making the changes and before making the amend commit

**Comment by jonathan-grah** [comment]
Thanks for contributing to freeCodeCamp @cpheimbach 😄 This is being overhauled in my pull request (#16185). Let's merge that and then we can see from there. Hopefully you can find another issue that has the label [help wanted](https://github.com/freeCodeCamp/freeCodeCamp/labels/help%20wanted) ❗️

**Comment by raisedadead** [comment]
Hi @jonathan-grah 

Your pull request (https://github.com/freeCodeCamp/freeCodeCamp/pull/16185) did not add the buttons to the static profile page. And was merged.

You have made another PR https://github.com/freeCodeCamp/freeCodeCamp/pull/16186 for react profiles, which is under QA and not merged.

Hence your assumption that it will be merged and everything will work is NOT correct.

As of this way there is no way to confirm if the certs work fine from a UAT point of view (which is how a user would see them).

We should NOT assume something will be fixed, and always have the simplest path update the code-base iteratively. 

/cc @BerkeleyTrue @QuincyLarson

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
Code LGTM. Thanks @cpheimbach 

@BerkeleyTrue if we have any reason not to merge this, please go ahead and close.

**Comment by jonathan-grah** [comment]
@raisedadead; I have been talking to @QuincyLarson about this matter, and we decided it was best to close this issue, as it wasn't worth merging something only for it to be overwritten in 1-2 days. The branch `staging`, as far as I know, is not being pushed to the beta release for a day or two.

**Comment by raisedadead** [comment]
I'll be blunt and honest in my views (although this is not in scope of this thread) :

> I have been talking to @QuincyLarson about this matter, and we decided it was best to close this issue, as it wasn't worth merging something only for it to be overwritten in 1-2 days. 

Sure, then my concern is why wasn't it made public? Why do we need to make assumptions and break things?

Were all of the backend routes tested by some one other than the one who wrote the code, when your original PR was merged?

Why is it not a good idea to have a contribution that (even though to be removed later) which brings in new contributors?
  
> The branch staging, as far as I know, is not being pushed to the beta release for a day or two.

That proves the point that its not ready for deployment. And the whole point of having a `staging` branch it is to be deploy-able for UAT at any given point of time.

And, hence calling out anyone to test and fix things makes sense.

Else, its just a loop of breaking and fixing and breaking, and NOT iterative development. 

----

P.S:

Let's be careful, its easy to motivate anyone contribute, its easier to discourage them. It takes one comment, action etc, to break that trust.

Note:  _"With great power comes great responsibility"_

Every now and then say we see issues which are opened as regressions, when we overhaul something, simply because we neglect to follow established best practices.

We see lots of new contributors, and only a handful come back again because of the lack of clarity, and this is apparent to me over the couple years that I have seen this code-base.

Check the statics, on how many contributions go into `/seed` and how many go to other areas.

I am truly committed to this community, and help in bring more contributors to core areas of the code-base. But discussions and philosophies likes of which are happening here in the thread, sour my motivation to do so.

Nonetheless, that's my two cents. 

Feel free to merge or close the PR as decided.

**Comment by jonathan-grah** [comment]
@raisedadead; I understand your position, I did mention that it isn't the best practice to merge an unfinished pull request into `staging`, and leave it that way. I am completely aware of these standards, and usually completely abide by them. In this case, Quincy gave me the heads up to go ahead and close this pull request, as we are focusing on overhauling this instead.

I'm going to merge this pull request, and I do agree with your reasons for wanting to merge it. However, in future, if any of us have issues regarding a closed PR or issue, we should contact the person directly instead of commenting on GitHub 👍 

*Were all of the backend routes tested by someone other than the one who wrote the code, when your original PR was merged?* Yes, as far as I know

*Why is it not a good idea to have a contribution that (even though to be removed later) which brings in new contributors?* I understand this, but I think your blunt messages have probably had more of an impact. I think in future this should be done in private messages.

**Comment by jonathan-grah** [comment]
Thank @cpheimbach for your contribution, sorry for the confusion!

**Comment by QuincyLarson** [comment]
@raisedadead To be clear, I QA'd this and merged it, and in doing so take responsibility for anything that it may have broken. 

There is a tension between getting code merged in early and often and trying to get code "right". In this case, since we're not deploying this directly production at this point and there's still time to refactor, I opted to merge.

**Comment by raisedadead** [comment]
@jonathan-grah with due respect this is not about this PR. And I sincerely apologise for my choice of words, but the concerns are still the same.

@QuincyLarson thanks a lot, but no one including you is responsible for anything broken. It should be as simple as following iterative development.

Again, I apologise if my choice of words sounded like a rant.


Let me clarify as I see I may not have been clearer earlier. 

- I noticed something which can be a contribution, and created an issue for some one to contribute.
- A contributor came in a made a PR, which was closed. The comments that were left were inadequate even for me who knows about whats going on with the code-base, lately.
- The clarification was made only after, I jumped in and re-opened this pull request.

My concerns were not just about one single PR, its the practice in general that I have seen and have mentioned in my comments above.

I understand that some discussions can and will happen one-to-one and thats all fine by me.

But I care when that isn't clarified to all contributors. We closed the Core Chat to have more involved public discussions in the Contributors room.

In any case, I will this keep in mind in future, while reporting/triaging issues and reviewing PRs, which I would err on the side of until beta is shipped.

Lets focus on what we are good at, fixing things and not getting in debates.
Happy contributing!

**Comment by QuincyLarson** [comment]
@raisedadead Thanks for the additional clarification. Yes - when we closed that PR you're talking about, @jonathan-grah and I were on the phone discussing it. We should have written a more comprehensive explanation of why we were closing it.

**Pull request merged**